torch/csrc/utils/tensor_new.cpp

```diff
@@ -270,7 +270,7 @@ Tensor internal_new_from_data(
 
   if (PyArray_Check(data)) {
     TORCH_CHECK(!pin_memory, "Can't pin tensor constructed from numpy");
-    auto tensor = tensor_from_numpy(data);
+    auto tensor = tensor_from_numpy(data, copy_numpy);
```

Comment:
`copy_numpy` -> `!copy_numpy`?

Reply:
That is, you want to `warn_if_not_writeable` when this does not copy, right?

Reply:
Updated in https://github.com/pytorch/pytorch/pull/47271/commits/c9a7efa43c9dc406335e0b6beda757228f9599a9